contrib/gocql/gocql/observer.go

```diff
@@ -0,0 +1,123 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024 Datadog, Inc.
+
+package gocql
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	"github.com/gocql/gocql"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// NewTracedSession returns a new session augmented with tracing.
+func NewTracedSession(cluster *gocql.ClusterConfig, opts ...WrapOption) (*gocql.Session, error) {
```

Comment:
Question about the naming here -> `NewTracedSession` or `CreateTracedSession` or `TraceCreateSession`...

The idea is that initially, `gocql` uses `Create` and not `New`. `New` sounds like "it should not fail, not return an error" as in, returns an object, whatever. Here we're creating something, it may fail, it may succeed...

I don't know, maybe I am nitpicking, but just wanted to share the idea. At the same time I acknowledge we already have `gocqltrace.CreateSession` so does `gocqltrace.CreateTracedSession` make sense? Wanted to share the thought, though.

Maybe `gocqltrace.CreateSessionWithObserver` ? (naming is hard...)

Reply:
@ufoot Piggybacking on the "observer" theme, we could name it `gocqltrace.CreateObservedSession` 😁

Reply:
Thanks for the suggestions. I will go with the suggested `gocqltrace.CreateTracedSession` as I would like to keep the word "tracing" around 😄.